crates/shared/src/signature_validator.rs

```diff
@@ -86,6 +99,27 @@ impl SignatureValidating for Web3SignatureValidator {
         batch.execute_all(MAX_BATCH_SIZE).await;
         future::join_all(calls).await
     }
+
+    async fn validate_signature_and_get_additional_gas(
+        &self,
+        check: SignatureCheck,
+    ) -> Result<U256, SignatureValidationError> {
+        let instance = ERC1271SignatureValidator::at(&self.web3, check.signer);
+        let is_valid_result = instance
+            .is_valid_signature(Bytes(check.hash), Bytes(check.signature.clone()))
+            .call()
+            .await;
+
+        let is_valid_gas_estimate = instance
+            .is_valid_signature(Bytes(check.hash), Bytes(check.signature))
```

Comment:
Stupid question. Does this contain the 21k gas tx initiation cost? Because our interactions don't have the 21k gas overhead?

Reply:
I honestly don't know. It's using the https://docs.rs/ethcontract/latest/ethcontract/transaction/struct.TransactionBuilder.html#method.estimate_gas to estimate the gas and from what I can see it's doing an `estimateGas` call

Reply:
Yes, I think we can subtract the 21k, but have to add around 2600 for the call into a new contract:
![image](https://user-images.githubusercontent.com/7348154/196497015-6f1dea21-4e6f-4513-a4af-840f28fdb4c8.png)
I check the gas costs always over here:
https://github.com/wolflo/evm-opcodes/blob/main/gas.md#a5-balance-extcodesize-extcodehash

Of course, 21k - 2.6k is a super rough estimate. But I think its good enough for us.

Reply:
Good point, we need to account for the access cost!

Reply:
**Edit**: On second, thought, we can be more generous and subtract the full 21K from the gas estimate. This is because we save the 3000K from calling the `ecrecover` precompile compared to ECDSA-signed orders.